### PR TITLE
Make StringC.toDString more D2 friendly

### PR DESCRIPTION
**Pull request by mathias-lang-sociomantic**
CC @jens-mueller-sociomantic

### CODE REVIEW

File: src/ocean/text/util/StringC.d [hunk 1]
```diff
@@ -133,9 +133,9 @@ class StringC
 
     ***************************************************************************/
 
-    public static cstring toDString ( char* str )
+    public static Inout!(char)[] toDString ( Inout!(char)* str )
     {
-        return str ? str[0 .. strlen(str)] : "";
+        return str ? str[0 .. strlen(str)] : str[0 .. 0];
```
Comment (mihails-strasuns-sociomantic) [inline src/ocean/text/util/StringC.d]: ` : null` is probably more straightforward

Comment (mathias-lang-sociomantic) [reply src/ocean/text/util/StringC.d]: Fixed

**Review by mihails-strasuns-sociomantic — APPROVED** [review]

**Comment by mihails-strasuns-sociomantic** [comment]
Rebase on 2.5.x ?

**Comment by nemanja-boric-sociomantic** [comment]
Why this doesn't target v2.5.x?

**Comment by mathias-lang-sociomantic** [comment]
Done

**Review by nemanja-boric-sociomantic — APPROVED** [review]

**Comment by nemanja-boric-sociomantic** [comment]
I've changed the base to v2.5.x

**Pull request merged**